### PR TITLE
refactor(unlock-assets): Generate typescript components

### PR DESCRIPTION
**Pull request by jordang7**
# Description

Modified unlock-assets package to have svgr generate typescript components
- Removed js files in build/svg-components and replaced with typescript
- Added interfaces variable to template.js to allow for typescript and --typescript flag added to svg-2-components script

# Issues

#8851 

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by jordang7** [comment]
> Can you run `unlock-app` after rebuilding everything locally and see if the icons are displayed properly on checkout, dashboard, etc?

Yes just did a packages:clean and build and the icons look to be rendering properly. 
Screenshot from 2022-08-29 10-49-17

**Comment by jordang7** [comment]
Secondly, while we are generating typescript components for build folder. We aren't actually outputing them in dist when we use bundler to export them. Please follow the unlock-js build command and replace micro bundler here with what we are using there to output the typescript definition as well.

Just replaced microbundler with tsup. However I can't seem to replicate the watermark not displaying. It's showing on the demo page correctly for me. 
unl

Could you please send me the steps you took to produce the bug? Thanks!

File: packages/unlock-assets/package.json [hunk 1]
```diff
@@ -5,11 +5,11 @@
   "author": "unlock inc",
   "license": "MIT",
   "main": "dist/index.js",
-  "module": "dist/index.modern.js",
-  "source": "src/index.js",
+  "module": "dist/index.mjs",
+  "source": "src/index.tsx",
```
Comment (searchableguy) [inline packages/unlock-assets/package.json:9]: add `typings` property referencing the ts definition file.

File: packages/unlock-assets/package.json [hunk 2]
```diff
@@ -5,11 +5,11 @@
   "author": "unlock inc",
   "license": "MIT",
   "main": "dist/index.js",
-  "module": "dist/index.modern.js",
-  "source": "src/index.js",
+  "module": "dist/index.mjs",
```
Comment (searchableguy) [inline packages/unlock-assets/package.json:8]: Use `exports` instead of module defined on the node official documentation site.

**Comment by searchableguy** [comment]
Reproduction steps:

```shell
git pull jordan branch
cd packages/unlock-assets
yarn install
yarn svg-2-components
yarn build
cd ../../unlock-app
yarn dev
```

The icon does not show up for me. Some are missing.

**Comment by jordang7** [comment]
> CleanShot 2022-09-10 at 02 59 52@2x
> 
> There are some icons which have changed and aren't appearing. This is from the checkout bottom - unlock word mark.
> 
> Can you check and fix what caused this difference after rebuilding?

Did a total clean/rebuild and I still haven't been able to replicate this bug on my end so haven't been able to find a fix. This screenshot is from the demo feature on the unlock-app dashboard, correct? Or is there another way you are reaching this checkout page? Also are there any console errors when seeing this bug?

**Comment by searchableguy** [comment]
Hm, that's very odd. Can you tell me your reproduction steps, browser environment, and OS?

**Comment by jordang7** [comment]
Ah yes that was it!

I can see the bug now, thanks. Will take a look at figure out what's happening.

**Comment by searchableguy** [comment]
is this fixed?

**Comment by jordang7** [comment]
Not fully. I fixed the way it displays on the checkout page but I realize it broke the way it displays on the dashboard if forContent is true (shows black instead of brand color). Need to look into it a little more

**Comment by jordang7** [comment]
Yes should be fixed now

**Review by searchableguy — APPROVED** [review]
Approving. I've made some changes to the template. 

Though type checking isn't available in the unlock-app package right now due to vendor.d.ts and there are some changes required there but that's out of scope here.

**Pull request merged**